tests/lib/bin/version-tool

```diff
@@ -0,0 +1,323 @@
+        self.assertEqual(naive_version_cmp("1", "1.2"), -1)
+        self.assertEqual(naive_version_cmp("1.2", "1"), +1)
+        self.assertEqual(naive_version_cmp("1", "1.0.1"), -1)
+        self.assertEqual(naive_version_cmp("1.1", "1.0.1"), +1)
+
+    def test_version_with_text(self):
+        # type: () -> None
+        self.assertEqual(naive_version_cmp("1-foo", "1-bar"), 0)
+        self.assertEqual(naive_version_cmp("1-foo.2-froz", "1-bar"), +1)
+        self.assertEqual(naive_version_cmp("1-foo.2-froz", "1-bar.3-quux"), -1)
+
+    def test_kernel_version(self):
+        # type: () -> None
+        self.assertEqual(naive_version_cmp("5.2.9-1-default", "4.13"), +1)
+
+
+def _make_parser():
+    # type: () -> ArgumentParser
+    parser = ArgumentParser(
+        epilog="""
+Relational operator is one of:
+
+    -eq -ne -gt -ge -lt -le
+
+Version comparison is performed using the selected algorithm.
+
+naive:
+    The algorithm considers each version to be a tuple of integers. Non,
+    integer elements or element fragments are discarded.
+
+    Comparison is performed on by considering the leftmost element in each
+    tuple. First pair of numbers that are not equal determine the result of the
+    comparison. Tuples have unequal length then missing elements are
+    substituted with zero.
+    """,
+        formatter_class=RawTextHelpFormatter,
+    )
+    parser.register("action", "unit-test", _UnitTestAction)
+    parser.add_argument("-v", "--version", action="version", version="1.0")
+    parser.add_argument(
+        "--verbose", action="store_true", help="describe comparison process"
+    )
+
+    parser.add_argument("version_a", metavar="VERSION-A")
+    parser.add_argument("version_b", metavar="VERSION-B")
+
+    # algorithm selection
+    alg_grp = parser.add_mutually_exclusive_group(required=True)
+    alg_grp.add_argument(
+        "--naive",
+        dest="algorithm",
+        action="store_const",
+        const="naive",
+        help="select the naive version comparison",
+    )
+    # relation selection
+    rel_op_grp = parser.add_mutually_exclusive_group(required=True)
+    rel_op_grp.add_argument(
+        "-eq",
+        action="store_const",
+        const="==",
+        dest="rel_op",
+        help="test that versions are equal",
+    )
+    rel_op_grp.add_argument(
+        "-ne",
+        action="store_const",
+        const="!=",
+        dest="rel_op",
+        help="test that versions are not equal",
+    )
+    rel_op_grp.add_argument(
+        "-gt",
+        action="store_const",
+        const=">",
+        dest="rel_op",
+        help="test that version-a is greater than version-b",
+    )
+    rel_op_grp.add_argument(
+        "-ge",
+        action="store_const",
+        const=">=",
+        dest="rel_op",
+        help="test that version-a is greater than or equal to version-b",
+    )
+    rel_op_grp.add_argument(
+        "-lt",
+        action="store_const",
+        const="<=",
```

Comment:
this should be `<`
```suggestion
        const="<",
```

Reply:
This made me realise we're missing one last layer of tests: spread. Added now, will push after. local run.